palantir-java-format/src/main/java/com/palantir/javaformat/java/java14/Java14InputAstVisitor.java

```diff
@@ -0,0 +1,237 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.javaformat.java.java14;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.MoreCollectors.toOptional;
+import static com.sun.source.tree.Tree.Kind.BLOCK;
+
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.palantir.javaformat.Op;
+import com.palantir.javaformat.OpsBuilder;
+import com.palantir.javaformat.java.JavaInputAstVisitor;
+import com.sun.source.tree.BindingPatternTree;
+import com.sun.source.tree.CaseTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.SwitchExpressionTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.YieldTree;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.tree.JCTree;
+import com.sun.tools.javac.tree.TreeInfo;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extends {@link JavaInputAstVisitor} with support for AST nodes that were added or modified for Java 14.
+ */
+public class Java14InputAstVisitor extends JavaInputAstVisitor {
+
+    public Java14InputAstVisitor(OpsBuilder builder, int indentMultiplier) {
+        super(builder, indentMultiplier);
+    }
+
+    @Override
+    public Void visitBindingPattern(BindingPatternTree node, Void unused) {
+        sync(node);
+        scan(node.getType(), null);
+        builder.breakOp(" ");
+        visit(node.getBinding());
+        return null;
+    }
+
+    @Override
+    public Void visitYield(YieldTree node, Void aVoid) {
+        sync(node);
+        token("yield");
+        builder.space();
+        scan(node.getValue(), null);
+        token(";");
+        return null;
+    }
+
+    @Override
+    public Void visitSwitchExpression(SwitchExpressionTree node, Void unused) {
+        sync(node);
+        visitSwitch(node.getExpression(), node.getCases());
+        return null;
+    }
+
+    @Override
+    public Void visitClass(ClassTree tree, Void unused) {
+        switch (tree.getKind()) {
+            case ANNOTATION_TYPE:
+                visitAnnotationType(tree);
+                break;
+            case CLASS:
+            case INTERFACE:
+                visitClassDeclaration(tree);
+                break;
+            case ENUM:
+                visitEnumDeclaration(tree);
+                break;
+            case RECORD:
+                visitRecordDeclaration(tree);
+                break;
+            default:
+                throw new AssertionError(tree.getKind());
+        }
+        return null;
+    }
+
+    public void visitRecordDeclaration(ClassTree node) {
```

Comment:
this is pretty hard to read, but it looks like at least it matches the current g-j-f impl, so at least we'll be able to stay in sync. https://github.com/google/google-java-format/blob/93d649d0fbc07fc604f7b47b3098cf6e1df521de/core/src/main/java/com/google/googlejavaformat/java/java14/Java14InputAstVisitor.java#L99-L160